cmd/snap-confine/mount-support.h

```diff
@@ -33,4 +33,6 @@
  **/
 void sc_populate_mount_ns(const char *snap_name);
 
+/** Ensure that /snap is mounted with the SHARED option */
+void sc_ensure_shared_snap_mount();
```

Comment:
Can you please expand this to document how this works. It does the check on `/` and `/snap` and this is not mentioned.